docs/docs/development/hardware-integration/index.mdx

```diff
@@ -0,0 +1,122 @@
+
+### What is a "shield"?
+
+In ZMK, a _shield_ is a _PCB_ or _hardwired set of components_ that when combined with an MCU-only board, like the SparkFun Pro Micro RP2040 or nice!nano, results in a complete usable keyboard.
+Examples would be keyboard PCBs like the Kyria or Lily58.
+he shield is usually the big PCB containing all the keys.
+
+## Organization Overview
+
+<Tabs
+defaultValue="self-contained"
+values={[
+{label: 'Self-contained keyboards', value: 'self-contained'},
+{label: 'Composite keyboards', value: 'composite'},
+]}>
+
+<TabItem value="self-contained">
+
+For a [self-contained keyboard](../../hardware.mdx#onboard) that includes the microprocessor, all of the above architecture components are included in the Zephyr _board_ definition and no shield is defined.
+You can see an example for the [Planck V6](https://github.com/zmkfirmware/zmk/tree/main/app/boards/arm/planck) board directory.
+
+With this type of keyboard, the full ZMK definition for the keyboard exists in the `<board_root>/boards/<arch>/<keyboard_name>` directory where `<board_root>` is `zmk/app` or a [module](../../features/modules.mdx) root, e.g. `zmk/app/boards/arm/planck/`.
+In that directory you'll have the following files, where there can be multiple `<board_name>`s, corresponding to each keyboard part for [split keyboards](../../features/split-keyboards.md):
+
+- A `Kconfig.board` file that defines the toplevel [Kconfig](https://docs.zephyrproject.org/3.5.0/build/kconfig/index.html) items for the board, including which SoC Kconfig setting it depends on.
+- A `Kconfig.defconfig` file that sets some initial defaults when building this keyboard. This usually includes:
+  - Setting `ZMK_KEYBOARD_NAME` to a value, for the product name to be used for USB/BLE info.
+  - Setting `ZMK_USB` and/or `ZMK_BLE` for the default values for which HID transport(s) to enable by default
+- A `<board_name>_defconfig` file that forces specific Kconfig settings that are specific to this hardware configuration.
+  These are mostly SoC settings around the specific hardware configuration.
+- `<board_name>.dts` which contains all the devicetree definitions[^1], including but not limited to:
+  - An `#include` line that pulls in the specific microprocessor that is used, e.g. `#include <st/f3/stm32f303Xc.dtsi>`.
+  - Kscan, matrix transform and physical layout devicetree nodes as described above
+  - A [chosen](https://docs.zephyrproject.org/3.5.0/build/dts/intro-syntax-structure.html#aliases-and-chosen-nodes) node including `zmk,physical-layout` property among others, where each property references the nodes defined in the file.
+- A `board.cmake` file with CMake directives for how to flash to the device.
+- A `<keyboard_name>.keymap` file that includes the default keymap for that keyboard. Users will be able to override this keymap in their user configs.
+- A `<keyboard_name>.zmk.yml` file containing [metadata](hardware-metadata-files.md) for the keyboard.
```

Comment:
This all still feels very verbose, and it also feels somewhat duplicatey. Perhaps separating out the kconfig, devicetree, and the other files would help? Also wonder if it might be better to link to config sections for things like `ZMK_KEYBOARD_NAME`.

Reply:
Yeah, it seems sensible to split these by category. I'll add a link to the config page for those as well.

Reply:
Split them and categorized, looks better now IMO.